packages/cardano-services/src/Asset/types.ts

```diff
@@ -16,3 +16,15 @@ export interface NftMetadataService {
    */
   getNftMetadata(asset: AssetPolicyIdAndName): Promise<Asset.NftMetadata | undefined>;
 }
+
+/**
+ * Service to get CIP-35 token metadata for a given subject
+ */
+export interface TokenMetadataService {
+  /**
+   * Get CIP-35 token metadata for a given subject
+   *
+   * @returns CIP-35 token metadata, `null` if not found
+   */
+  getTokenMetadata(subject: string): Promise<Asset.TokenMetadata | null>;
```

Comment:
```suggestion
  getTokenMetadata(assetIds: AssetId[]): Promise<Asset.TokenMetadata[] | null>;
```
The Cardano Token Registry, on which we're basing our interface, supports batch fetching in a single request.